cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -512,15 +512,34 @@ macro(build_boost)
   set(BOOST_BUILD_PRODUCTS ${BOOST_STATIC_SYSTEM_LIBRARY}
                            ${BOOST_STATIC_FILESYSTEM_LIBRARY}
                            ${BOOST_STATIC_REGEX_LIBRARY})
-  set(BOOST_CONFIGURE_COMMAND "./bootstrap.sh" "--prefix=${BOOST_PREFIX}"
-                              "--with-libraries=filesystem,regex,system")
+  if(MSVC)
+    set(BOOST_CONFIGURE_COMMAND "bootstrap.bat" "--prefix=${BOOST_PREFIX}"
+                                "--with-libraries=filesystem,regex,system")
+  else()
+    set(BOOST_CONFIGURE_COMMAND "./bootstrap.sh" "--prefix=${BOOST_PREFIX}"
+                                "--with-libraries=filesystem,regex,system")
+  endif()
+
   if("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
     set(BOOST_BUILD_VARIANT "debug")
   else()
     set(BOOST_BUILD_VARIANT "release")
   endif()
-  set(BOOST_BUILD_COMMAND "./b2" "link=${BOOST_BUILD_LINK}"
-                          "variant=${BOOST_BUILD_VARIANT}" "cxxflags=-fPIC")
+  if(MSVC)
+    set(BOOST_BUILD_COMMAND
+        "b2.exe"
+        "link=${BOOST_BUILD_LINK}"
+        "variant=${BOOST_BUILD_VARIANT}"
+        "address-model=64"
+        "toolset=msvc-14.0"
```

Can the MSVC version need to match the current toolset in use? If we are using vs2017 or vs2019 then the toolset version number probably needs to be correspondingly higher here

---

CMake has a built-in variable `MSVC_TOOLSET_VERSION` which can help here
```suggestion
  if(MSVC)
    if(MSVC_TOOLSET_VERSION EQUAL 142)
      set(TOOLSET "msvc-14.2")
    elseif(MSVC_TOOLSET_VERSION EQUAL 141)
      set(TOOLSET "msvc-14.1")
    elseif(MSVC_TOOLSET_VERSION EQUAL 140)
      set(TOOLSET "msvc-14.0")
    else()
      message(SEND_ERROR "unsupported MSVC toolset version, MSVC_TOOLSET_VERSION=${MSVC_TOOLSET_VERSION}") 
    endif()
```